Buildings/Controls/OBC/Utilities/PIDWithAutotuning/FirstOrderAMIGO.mo

```diff
@@ -0,0 +1,322 @@
+  connect(PIPar.k, samk.u) annotation (Line(points={{-82,86},{-94,86},{-94,-20},
+          {-42,-20}}, color={0,0,127}));
+  connect(PIPar.Ti, samTi.u) annotation (Line(points={{-82,74},{-90,74},{-90,
+          -50},{-82,-50}}, color={0,0,127}));
+  connect(resPro.triEnd, conProMod.triEnd) annotation (Line(points={{-2,32},{-36,
+          32},{-36,38}}, color={255,0,255}));
+  connect(resPro.triSta, conProMod.triSta) annotation (Line(points={{-2,36},{-24,
+          36},{-24,38}}, color={255,0,255}));
+  connect(resPro.triEnd, samTi.trigger) annotation (Line(points={{-2,32},{-10,
+          32},{-10,10},{-70,10},{-70,-38}}, color={255,0,255}));
+  connect(resPro.triEnd, samk.trigger) annotation (Line(points={{-2,32},{-10,32},
+          {-10,10},{-30,10},{-30,-8}},color={255,0,255}));
+  connect(resPro.triEnd, samTd.trigger) annotation (Line(points={{-2,32},{-10,
+          32},{-10,-44},{-30,-44},{-30,-58}}, color={255,0,255}));
+  connect(PIDPar.Td, samTd.u) annotation (Line(points={{-82,43},{-86,43},{-86,-70},
+          {-42,-70}}, color={0,0,127}));
+  connect(swi.y, y)
+    annotation (Line(points={{82,-20},{90,-20},{90,0},{120,0}}, color={0,0,127}));
+  connect(u_m, PID.u_m) annotation (Line(points={{0,-120},{0,-96},{10,-96},{10,
+          -52}}, color={0,0,127}));
+  connect(rel.trigger, triTun) annotation (Line(points={{24,-2},{24,-82},{60,-82},
+          {60,-120}}, color={255,0,255}));
+  connect(or1.u1, resPro.triEnd) annotation (Line(points={{58,-60},{52,-60},{52,
+          -50},{46,-50},{46,-20},{-10,-20},{-10,32},{-2,32}}, color={255,0,255}));
+  connect(or1.y, swi.u2) annotation (Line(points={{82,-60},{90,-60},{90,-40},{
+          48,-40},{48,-20},{58,-20}}, color={255,0,255}));
+  connect(not1.u, triTun) annotation (Line(points={{28,-68},{24,-68},{24,-82},{60,
+          -82},{60,-120}}, color={255,0,255}));
+  connect(not1.y, or1.u2)
+    annotation (Line(points={{52,-68},{58,-68}}, color={255,0,255}));
+  annotation (Documentation(info="<html>
+<p>
+This block implements a rule-based PID tuning method.
+Specifically, this PID tuning method approximates the control process with a
+first-order time delayed (FOTD) model.
+It then determines the parameters of this FOTD model based on the responses of
+the control process to an asymmetric relay feedback.
+After that, taking the parameters of this FOTD mode as inputs, this PID tuning
+method calculates the PID parameters with an Approximate M-constrained Integral Gain Optimization (AMIGO) Tuner.
+This block is built based on
+<a href=\"modelica://Buildings.Controls.OBC.Utilities.PIDWithInputGains\">
+Buildings.Controls.OBC.Utilities.PIDWithInputGains</a>
+and inherits most of the parameters of the latter. However, through the parameter
+<code>controllerType</code>, the controller can only be configured as PI or
+PID controller.
+In addition, the output of this block is limited from 0 to 1.
+</p>
+<p>
+Note that the autotuning can be performed only once.
```

Comment:
I limit it to be only once to avoid the following scenario:
- the autorunning has not been completed yet but the <code>triTun</code> becomes True.

Please let me know your thoughts on it.

Reply:
One scenario may need to double check:
- [ ] If `triTun` is `true` and `resPro.triEnd` is `false` (means autotuning has not been completed yet but the `triTun` becomes True), then `or1.y` will be `false`. The `y` will be `swi.u3`, which equals to `rel.y`. However, the `rel.y` would not equal to `yHig`, in block `PIDWithAutotuning.Relay.Controller`, as the `trigger` input in the block is `true`.

Reply:
Yes, <code>rel.y</code> may not be equal to <code>yHig</code>.
Can you please let me know if <code>rel.y</code> needs to be equal to <code>yHig</code>?

Reply:
Is there a scenario that the `triTun` becomes `false` but the `resPro.triEnd` is `false` (means the autotuning triggering signal becomes `false` but the tuning has not been completed), or anytime when the `triTun` becomes `false`, the `resPro.triEnd` will become `true`?
Is it fine to change the text as below?
```
Note that each time when the tuning triggering input `triTun` becomes `true`, the autotuning can be performed and performs only one time.
```

Reply:
The current implementation assumes that <code>triTun</code> keeps <code>true</code> after it becomes <code>true</code>.
The scenario you described is unexpected.
The text looks good but do we need to emphasize that <code>triTun</code> needs to keep <code>true</code>?

Reply:
so will it be possible to retune the controller in the middle of the system operation? what is the typical scenario that the controller should be tuned? I basically wants to understand where the input `triTun` receives the signal.

Reply:
Retuning is possible only when the ongoing tuning is completed.
That means when we can retune can't be predefined.
We could add a Boolean output to indicate if the retuning window opens.

I thought the main reason to add this <code>triTun</code> is to avoid tuning this PI during the unoccupied period.
In this case, the <code>triTun</code> should be connected with the occupied signal.
Please let me know your thoughts.

Reply:
Is it possible to re-trigger the tuning process? For current implementation, at the first time when `triTun` changes from `false` to `true`, the `resPro.triEnd` is `false` (is it correct?), then the `swi.y` equals to `rel.y`, which I think the tuning process starts. 
After the tuning is done, the `resPro.triEnd` becomes `true` and keeps being `true`. And then the `triTun` changes to be `false`. (Will this make the `resPro.triEnd` becomes `false`?)
If the `resPro.triEnd` keeps being `true`, it will be not possible to re-trigger the tuning process.

Reply:
Yes, you are right. The current implementation doesn't allow to re-trigger the tuning process. I didn't fix this issue mainly because we couldn't predict when the tuning is done. I thought it basically eliminated the possibility for performing re-tuning.

Reply:
The implementation must allow re-tuning. We want to be able to trigger it from an external block whenever a control loop has poor performance.